src/parser/ParsedQuery.cpp

```diff
@@ -405,48 +412,89 @@ ParsedQuery::GraphPatternOperation::GraphPatternOperation(
+    case Type::SUBQUERY:
+      _subquery = nullptr;
+      break;
+    default:
+      AD_THROW(ad_semsearch::Exception::CHECK_FAILED,
+               "This constructor should only be used for SUBQUERY"
+               "type operations.");
+  }
+}
+
 // _____________________________________________________________________________
 ParsedQuery::GraphPatternOperation::GraphPatternOperation(
     GraphPatternOperation&& other)
-    : _type(other._type),
-      _childGraphPatterns(std::move(other._childGraphPatterns)) {
-  other._childGraphPatterns.clear();
+    : _type(other._type) {
+  if (_type == Type::SUBQUERY) {
+    _subquery = other._subquery;
+    other._subquery = nullptr;
+  } else {
+    new (&_childGraphPatterns) std::vector<GraphPattern*>();
+    _childGraphPatterns = std::move(other._childGraphPatterns);
+    other._childGraphPatterns.clear();
+  }
 }
 
 // _____________________________________________________________________________
 ParsedQuery::GraphPatternOperation::GraphPatternOperation(
     const GraphPatternOperation& other)
     : _type(other._type) {
-  _childGraphPatterns.reserve(other._childGraphPatterns.size());
-  for (const GraphPattern* child : other._childGraphPatterns) {
-    _childGraphPatterns.push_back(new GraphPattern(*child));
+  if (_type == Type::SUBQUERY) {
+    _subquery = new ParsedQuery(*other._subquery);
+  } else {
+    new (&_childGraphPatterns) std::vector<GraphPattern*>();
+    _childGraphPatterns.reserve(other._childGraphPatterns.size());
+    for (const GraphPattern* child : other._childGraphPatterns) {
+      _childGraphPatterns.push_back(new GraphPattern(*child));
+    }
   }
 }
 
 // _____________________________________________________________________________
 ParsedQuery::GraphPatternOperation& ParsedQuery::GraphPatternOperation::
 operator=(const ParsedQuery::GraphPatternOperation& other) {
-  _type = other._type;
-  for (GraphPattern* p : _childGraphPatterns) {
-    delete p;
+  if (_type == Type::SUBQUERY) {
+    delete _subquery;
+  } else {
+    for (GraphPattern* p : _childGraphPatterns) {
+      delete p;
+    }
   }
-  _childGraphPatterns.clear();
-  _childGraphPatterns.reserve(other._childGraphPatterns.size());
-  for (const GraphPattern* p : other._childGraphPatterns) {
-    _childGraphPatterns.push_back(new GraphPattern(*p));
+  _type = other._type;
+  if (_type == Type::SUBQUERY) {
+    _subquery = new ParsedQuery(*other._subquery);
+  } else {
+    _childGraphPatterns.clear();
+    _childGraphPatterns.reserve(other._childGraphPatterns.size());
+    for (const GraphPattern* p : other._childGraphPatterns) {
+      _childGraphPatterns.push_back(new GraphPattern(*p));
+    }
   }
   return *this;
 }
 
 // _____________________________________________________________________________
 ParsedQuery::GraphPatternOperation::~GraphPatternOperation() {
-  for (GraphPattern* child : _childGraphPatterns) {
-    delete child;
+  if (_type == Type::SUBQUERY) {
+    delete _subquery;
```

Comment:
`_subquery` is not a pointer‽ Does this work because of the placment new?

Reply:
`_subquery` should be a pointer.